Make HQQ default PTQ quantization in ExecuTorch

metascroy opened the pull request:
On MMLU, we see a 1% to 10% improvement on MMLU accuracy by using HQQ algorithm for quantization, depending on the model.

We make this the default quantization algorithm in the export LLM export script: 

```
MODEL			INT8-INT4		INT8-INT4-HQQ
---------------------------------------------
Gemma3-4B		0.50			0.55
SmolLM3-3B		0.5397			0.5546
Phi4-mini-4B	0.6050			0.6156
Qwen3-4B		0.6526			0.6585			
Llama1B			0.2892			0.3153	
```			

Evaulations were done using lm_eval with the following command:
```
HF_HUB_DISABLE_XET=1 lm_eval --model hf --model_args pretrained=quantized_model_id --tasks mmlu --device cuda:0 --batch_size 8
```

Here quantized_model_id corresponding to one of the above model/quant-scheme combinations uploaded to my huggingface account.

mergennachin reviewed (approved):
This is awesome!! Thanks for working on this

mergennachin commented:
Can we make this part of the release/1.0?

jackzhxng reviewed (commented):
How did you test gemma3 with etllm? (from your pr description)

metascroy committed:
Make HQQ default PTQ quantization in ExecuTorch (#14834)

Summary:
On MMLU, we see a 1% to 10% improvement on MMLU accuracy by using HQQ algorithm for quantization, depending on the model.

We make this the default quantization algorithm in the export LLM export script:

```
MODEL			INT8-INT4		INT8-INT4-HQQ
---------------------------------------------
Gemma3-4B		0.50			0.55
SmolLM3-3B		0.5397			0.5546
Phi4-mini-4B	0.6050			0.6156
Qwen3-4B		0.6526			0.6585
Llama1B			0.2892			0.3153
```

Evaulations were done using lm_eval with the following command:
```
HF_HUB_DISABLE_XET=1 lm_eval --model hf --model_args pretrained=quantized_model_id --tasks mmlu --device cuda:0 --batch_size 8
```

Here quantized_model_id corresponding to one of the above model/quant-scheme combinations uploaded to my huggingface account.

Pull Request resolved: https://github.com/pytorch/executorch/pull/14834

Reviewed By: mergennachin

Differential Revision: D84020605

Pulled By: metascroy
